modules/extreme_year.py

```diff
@@ -0,0 +1,117 @@
+    maximums = [[0, ""], [0, ""], [0, ""]]
+    initialized = False
+    # indexes to compare values with.
+    # highest temperature is stored at index 1 of weather reading
+    # lowest temperature is stored at index 3 of weather reading
+    # max humidity is stored at index 7 of weather reading
+    # see the weather files for further explanation
+    indexes = [1, 3, 7]
+    for i in files:
+        with open(i, "r") as file:
+            # skip first line as it contains field names
+            file.readline()
+            for line in file:
+                # since readline() returns 'linedata\n' with split by '\n' to get ['linedata','\n]
+                # then we get 'linedata' by grabbing the [0]
+                # lastly,  we split by ',' to get the entries parsed
+                parsed_line = line.split("\n")[0].split(",")
+                if not initialized:
+                    for j, val in enumerate(indexes):
+                        maximums[j] = [int(parsed_line[val]), parsed_line[0]]
+                    initialized = True
+                    continue
+                for j, val in enumerate(indexes):
+                    # index of weather record to compare value of current maximum with
+                    index_to_compare = val
+                    # value of temperatures or humidity or none
+                    value = parsed_line[index_to_compare]
+                    if value == "":
+                        continue
+                    value = int(value)
+                    if j != 1 and value >= maximums[j][0]:
+                        maximums[j] = [value, parsed_line[0]]
+                    elif value <= maximums[j][0]:
+                        maximums[j] = [value, parsed_line[0]]
+
+    return maximums
+
+
+def generate_extremes_report(maximums):
+    """
+    this functions prints/generates a report on the console based on the maximums array
+    the reports displays the following
+    1. max_highest_temperature with day
+    2. max_lowest_temperature with day
+    3. max_humidity with day
+    :param maximums:
+    :return: nothing
+    """
+    number_to_month = [
+        "January",
+        "February",
+        "March",
+        "April",
+        "May",
+        "June",
+        "July",
+        "August",
+        "September",
+        "October",
+        "November",
+        "December",
+    ]
+    initial_message = ["Highest", "Lowest", "Humidity"]
+    unit = ["C", "C", "%"]
+    for i, val in enumerate(maximums):
+        month, day = maximums[i][1].split("-")[1:]
+        print(
+            f"{initial_message[i]}: {str(val[0])}{unit[i]} "
+            f"on {number_to_month[int(month) - 1]} {day}"
+        )
+
+
+def extreme_temperatures_year(year, path):
+    """
+    this method uses two other methods to calculate max extreme temperatures and max humidity
+    for a given year
+    :param year: a string like: '2002', '2004', etc
+    :param path: a string like: 'weatherfiles/'
+    :return: nothing
```

Comment:
similar params documentation

Reply:
did u mean to document returns or something else?

Reply:
It would be something like below:

```
Args:
    Arg1 (type): brief detail
     Arg2 (type): brief detail

Returns:
    val1 (type): a brief detail
```

Reply:
done